src/toc.ts

```diff
@@ -0,0 +1,155 @@
+    .filter((file) => isValidFile(file) || isDirectory(file))
+    .sort()
+    .map((file) => {
+      if (isValidFile(file)) {
+        return {
+          file,
+          level,
+          ...fileInfo(file, fileSlugs),
+        } as LocalProjectPage;
+      }
+      const projectFolder = { title: fileInfo(file, fileSlugs).title, level };
+      const newLevel = level < 5 ? level + 1 : 6;
+      const pages = projectPagesFromPath(file, newLevel as pageLevels, fileSlugs, ignore);
+      return pages.length ? [projectFolder].concat(pages) : [];
+    })
+    .flat();
+}
+
+export function projectFromPath(path: string, indexFile?: string): LocalProject {
+  if (!indexFile) {
+    fs.readdirSync(path).forEach((file) => {
+      if (DEFAULT_INDEX_FILES.includes(file.toLowerCase())) {
+        indexFile = join(path, file);
+      }
+    });
+  }
+  if (!indexFile || !fs.existsSync(indexFile)) {
+    throw Error(`index file ${indexFile || DEFAULT_INDEX_FILES.join(',')} not found`);
+  }
+  const fileSlugs: Record<string, number> = {};
+  const { slug, title } = fileInfo(indexFile, fileSlugs);
+  const pages = projectPagesFromPath(path, 1, fileSlugs, [indexFile, join(path, '_build')]);
+  return { file: indexFile, index: slug, path, title, pages };
+}
+
+export function updateProject(store: Store<RootState>, path?: string, index?: string) {
+  path = path || '.';
+  const project = selectors.selectLocalProject(store.getState(), path);
+  if (!index && project?.file) {
+    index = project.file;
+  }
+  const newProject = projectFromPath(path, index);
+  store.dispatch(projects.actions.receive(newProject));
+}
+
+export function localToManifestProject(
+  proj: LocalProject,
+  projectSlug: string,
+  frontmatter: Frontmatter,
+): ManifestProject {
+  const { title: projectTitle, index, pages } = proj;
+  const manifestPages = pages.map((page) => {
+    if ('file' in page) {
+      const { slug, title, level } = page;
+      return { slug, title, level };
+    }
+    return page;
+  });
+  return { slug: projectSlug, index, title: projectTitle, pages: manifestPages, ...frontmatter };
+}
+
+function copyLogo(session: ISession, logoName?: string | null): string | undefined {
+  if (!logoName) {
+    session.log.debug('No logo specified, Curvespace renderer will use default logo');
+    return undefined;
+  }
+  if (!fs.existsSync(logoName)) {
+    // Look in the local public path
+    logoName = join('public', logoName);
+  }
+  if (!fs.existsSync(logoName))
+    throw new Error(`Could not find logo at "${logoName}". See 'config.web.logo'`);
+  const logo = `logo${extname(logoName)}`;
+  fs.copyFileSync(logoName, join(publicPath({}), logo));
+  return `/${logo}`;
+}
+
+export function getSiteManifest(session: ISession): SiteManifest {
+  const siteProjects: ManifestProject[] = [];
+  const state = session.store.getState();
+  const siteConfig = selectors.selectLocalSiteConfig(state);
+  if (!siteConfig) throw Error('no site config defined');
```

Comment:
Not sure if this is in progress or not, but this error was coming up a bunch on mine and I had no idea what to do. :)